haystack_experimental/dataclasses/tool.py

```diff
@@ -198,6 +203,66 @@ def get_weather(
 
         return Tool(name=name or function.__name__, description=tool_description, parameters=schema, function=function)
 
+    @classmethod
+    def from_component(cls, component: Component, name: str, description: str) -> "Tool":
+        """
+        Create a Tool instance from a Haystack component.
+
+        :param component: The Haystack component to be converted into a Tool.
+        :param name: Name for the tool.
+        :param description: Description of the tool.
+        :returns: The Tool created from the Component.
+        :raises ValueError: If the component is invalid or schema generation fails.
+        """
+
+        if not isinstance(component, Component):
+            raise ValueError(
+                f"{component} is not a Haystack component!" "Can only create a Tool from a Haystack component instance."
+            )
+
+        if getattr(component, "__haystack_added_to_pipeline__", None):
+            msg = (
+                "Component has been added in a Pipeline and can't be used to create a Tool. "
+                "Create Tool from a non-pipeline component instead."
+            )
+            raise ValueError(msg)
+
+        from haystack_experimental.components.tools.openai.component_caller import extract_component_parameters
```

Comment:
I have some doubts related to this line:
Why is this import placed here specifically?
Why is it in `[...].tools.openai.component_caller`? Is it compatible with OpenAI only or is it more generic?

Reply:
Yes, good point. I'll remove openai although one could argue this format is openai while Cohere for example has different format for tool calling definitions. Now, should we leave it in tools namespace with tool_invoker.py or perhaps move it perhaps close to tool.py in the same dataclasses module? The reason why I had it inside the method is circular deps which go away if we collocate `extract_component_parameters` next to tool.py in dataclasses. wdyt?